x-pack/plugins/file_upload/public/util/http_service.ts

```diff
@@ -6,15 +6,26 @@
  */
 
 import { i18n } from '@kbn/i18n';
+import { HttpFetchOptions } from 'kibana/public';
 import { getHttp } from '../kibana_services';
 
-export async function http(options) {
+export interface HttpOptions {
+  url: string;
+  method: string;
+  headers?: {
+    [key: string]: any;
+  };
+  data?: unknown;
+  query?: any;
+}
+
+export async function http(options: HttpOptions) {
```

Comment:
How about removing `http` and `doFetch` methods. `http` logic is not needed any more with the updates to the http service. That way, you can just call await fileUploadModules.getHttp().fetch directly from indexing_service.